File: swi-prolog-5.6.62/interpreter/length.pl

main :- 

 consult( testenv),

 checkComplainAndAdjustExpected( 0),

 ( length( [a, b, c], N1) -> 3=N1, testOk; testFail),
 ( length( L2, 4) -> length( L2, 4), testOk; testFail),

 checkComplainAndAdjustExpected( 2),

 % see also : 
 
 result(S), 
 format( S, []),

 true.

File: R-2.12.1/interpreter/length.R

source( 'testEnv.R')

te = TestEnv()

te$checkComplainAndAdjustExpected( 0)

te$test( 0 == length( logical( 0)))
te$test( 0 == length( numeric( 0)))
te$test( 0 == length( double( 0)))
te$test( 0 == length( integer( 0)))
te$test( 0 == length( character( 0)))

te$test( 1 == length( integer( 1)))
te$test( 2 == length( double( 2)))
te$test( 1 == length( 1))
te$test( 2 == length( c( 3, 4)))
te$test( 3 == length( c( 3, c( 4, 5))))

te$checkComplainAndAdjustExpected( 10)

# see also:

print( te$result())

